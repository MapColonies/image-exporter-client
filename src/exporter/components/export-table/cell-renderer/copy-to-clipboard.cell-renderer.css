span.buttonCopy{
  padding: 0.2em 0.5em;
  margin: 0 0.1em 0.1em 0.1em;
  border: 0.01em solid rgba(255,255,255,0);
  border-radius: 0.3em;
  box-sizing: border-box;
  text-decoration: none;
  font-weight: 300;
  color: #FFFFFF;
  text-shadow: 0 0.04em 0.04em rgba(0,0,0,0.35);
  text-align: center;
  transition: all 0.2s;
}
span.buttonCopy:hover{
  border-color: rgba(255,255,255,1);
}
@media all and (max-width:30em){
  span.buttonCopy{
  display:block;
  margin:0.2em auto;
  }
}

.copyIndicator{
  font-size: 10px;
  height: 10px;
  position: absolute;
  line-height: 6px;
  margin: 0 30%;
}

input.linkControl{
  width: 100px;
  text-overflow: ellipsis;
  height: 24px;
  position: relative;
  top: -1px;
}
input.linkControl:focus {
  outline: unset;
}
