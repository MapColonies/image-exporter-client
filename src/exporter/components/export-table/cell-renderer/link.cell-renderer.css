a.buttonLink{
  padding: 0.2em 1em;
  margin: 0 0.1em 0.1em 0;
  border: 0.16em solid rgba(255,255,255,0);
  border-radius: 1em;
  box-sizing: border-box;
  text-decoration: none;
  font-weight: 300;
  color: #FFFFFF;
  text-shadow: 0 0.04em 0.04em rgba(0,0,0,0.35);
  text-align: center;
  transition: all 0.2s;
}
a.buttonLink:hover{
  border-color: rgba(255,255,255,1);
}
@media all and (max-width:30em){
  a.buttonLink{
  display:block;
  margin:0.2em auto;
  }
}
