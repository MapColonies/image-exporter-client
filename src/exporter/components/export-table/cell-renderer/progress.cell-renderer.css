.rendererLabel{
    font-size: 12px;
    position: absolute;
    left: 50%;
    top: -4px;
}
