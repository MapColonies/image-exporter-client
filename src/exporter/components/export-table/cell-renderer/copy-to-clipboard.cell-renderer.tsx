import React, { useState } from 'react';
import { ICellRendererParams } from 'ag-grid-community';
import { FormattedMessage } from 'react-intl';
import CopyToClipboard from 'react-copy-to-clipboard';
import { useTheme } from '@map-colonies/react-core';
import './copy-to-clipboard.cell-renderer.css';

export const CopyToClipboardRenderer: React.FC<ICellRendererParams> = (
  props
) => {
  const theme = useTheme();
  const [copied, setCopied] = useState<boolean>(false);
  const value = props.getValue() as string;

  if (!value) {
    return <></>; //''; // not null!
  }
  return (
    <>
      {copied ? (
        <div
          className="copyIndicator"
          style={{color: theme.secondary}}
        >
           <FormattedMessage id="export-table.copy-indicator.text" />
        </div>
      ) : null}
      <input
        tabIndex={-1}
        value={value}
        className="linkControl"
      ></input>
      <CopyToClipboard text={value} onCopy={(): void => setCopied(true)}>
        <span className="buttonCopy" style={{ backgroundColor: theme.primary }}>
          <FormattedMessage id="export-table.copy.text" />
        </span>
      </CopyToClipboard>
    </>
  );
};
