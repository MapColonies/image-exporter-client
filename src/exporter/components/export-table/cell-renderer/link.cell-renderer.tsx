import React from 'react';
import { ICellRendererParams } from 'ag-grid-community';
import { FormattedMessage } from 'react-intl';
import { useTheme } from '@map-colonies/react-core';
import './link.cell-renderer.css';
import { IExportTaskStatus } from '../../../models/exportTaskStatus';

export const LinkRenderer: React.FC<ICellRendererParams> = (
  props
) => {
  const value: string = (props.data as IExportTaskStatus).link; 
  const theme = useTheme();
  
  if (!value) {
    return <></>;//''; // not null!
  }
  return (
    <a href={value} target="_blank" rel="noopener noreferrer" className="buttonLink" style={{backgroundColor:theme.primary}}>
      <FormattedMessage id="export-table.link.text" />
    </a>
  );

};
