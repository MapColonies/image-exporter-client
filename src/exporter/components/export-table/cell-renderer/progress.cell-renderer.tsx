import React from 'react';
import { ICellRendererParams } from 'ag-grid-community';
import { LinearProgress, Typography } from '@map-colonies/react-core';
import { Box } from '@map-colonies/react-components';
import './progress.cell-renderer.css';
import { useState } from 'react';
import EXPORTER_CONFIG from '../../../../common/config';
import { IExportTaskStatus, ExportStatus } from '../../../models/exportTaskStatus';

const PEDING_PROGRESS = 0;
const ONE_HUNDRED = 100;
export const ProgressRenderer: React.FC<ICellRendererParams> = (
  props
) => {
  const [isRtl] = useState<boolean>(EXPORTER_CONFIG.I18N.DEFAULT_LANGUAGE === 'he' ? true : false);
  const value: number = (props.data as IExportTaskStatus).progress; 
  const status: string = (props.data as IExportTaskStatus).status;
  if (!value || status === ExportStatus.FAILED) {
    return <></>;//''; // not null!
  }
  const getProgressValue = ():number => {
    return value > PEDING_PROGRESS ? (value / ONE_HUNDRED) : value;
  }

  const getPercentageText = ():string => {
    return value > PEDING_PROGRESS ? `${value}%` : `${value * ONE_HUNDRED}%`;
  }

  return (
    <Box style={{display: 'flex', height: '40px', alignItems: 'center', direction: 'ltr'}}>
      <Typography use="body1" className={'rendererLabel'}>{getPercentageText()}</Typography>
      <LinearProgress progress={getProgressValue()} reversed={isRtl} />
    </Box>
  );

};
