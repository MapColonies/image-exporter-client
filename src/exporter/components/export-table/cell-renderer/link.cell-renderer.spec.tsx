import React from 'react';
import { shallow } from 'enzyme';
import { ICellRendererParams, Column, RowNode, GridApi, ColumnApi } from 'ag-grid-community';
import { LinkRenderer } from './link.cell-renderer';

/* eslint-disable */
const mockDataBase:ICellRendererParams = {
  value: '',
  valueFormatted: null,
  getValue: () => {},
  setValue: () => {},
  formatValue: () => {},
  data: {link: ''} as any,
  node: new RowNode(),
  colDef: {},
  column: new Column({},null,'link',false),
  $scope: null,
  rowIndex: 1,
  api: new GridApi(),
  columnApi: new ColumnApi(),
  context: null,
  refreshCell: () => {},
  eGridCell: document.createElement('span'),
  eParentOfValue: document.createElement('span'),
  addRenderedRowListener: () => {},
};
/* eslint-enable */

describe('AgGrid LinkRenderer component', () => {
  it('renders correctly', () => {
    const mockData = {
      ...mockDataBase
    };

    const wrapper = shallow(
      <LinkRenderer {...mockData} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('value of link rendered as href', () => {
    const value = 'http://kuku.com/1';
    const mockData = {
      ...mockDataBase,
      value,
      data: { link: value },
    };

    const wrapper = shallow(
      <LinkRenderer {...mockData} />
    );
    
    const linkContainer = wrapper.find('a');
    expect(linkContainer.props().href).toBe(value);
  });

  it('empty link value not renders an <a>', () => {
    const value = '';
    const mockData = {
      ...mockDataBase,
      value,
      data: { link: value },
    };

    const wrapper = shallow(
      <LinkRenderer {...mockData} />
    );
    
    const linkContainer = wrapper.find('a');
    expect(linkContainer).toEqual({});
  });

});
