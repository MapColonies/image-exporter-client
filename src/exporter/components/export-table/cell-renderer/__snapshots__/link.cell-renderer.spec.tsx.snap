// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgGrid LinkRenderer component renders correctly 1`] = `<Fragment />`;
